pkg/resource/version.go

```diff
@@ -26,6 +26,18 @@ var (
 
 const undefinedVersion = "undefined"
 
+// Value returns the underlying version number. Returns 0 if the version is undefined.
+func (v Version) Value() uint64 {
+	return pointer.SafeDeref(v.uint64)
+}
+
+// Next returns a new incremented version.
+func (v Version) Next() Version {
```

Comment:
is it expected to update version when the existing is `zero` i.e: undefined :thinking:

Reply:
As we discussed, `zero` means non-existent and undefined means "I don't know about its version". Calling `.Next()` here will create a new version object for you, which is incremented by one on the called version. If called version is undefined, it will be treated as zero here.